Source/Request.swift

```diff
@@ -1206,15 +1209,16 @@ public final class DataStreamRequest: Request {
     }
 
     func didReceive(data: Data) {
-        $streamMutableState.read { state in
+        $streamMutableState.write { state in
             if let stream = state.outputStream {
                 underlyingQueue.async {
                     var bytes = Array(data)
                     stream.write(&bytes, maxLength: bytes.count)
                 }
             }
-
-            underlyingQueue.async { state.streams.forEach { stream in stream.queue.async { stream.stream(data) } } }
+            state.numberOfExecutingStreams += state.streams.count
+            let localState = state
+            underlyingQueue.async { localState.streams.forEach { $0(data) } }
```

Comment:
Is the `localState` copy here just to ensure all streams are called in the `async` dispatch if one of the streams were removed in a separate call since it's no longer in the lock?

Reply:
No, it's to work around a language limitation that doesn't allow `inout` values to escape their calling context.